tests/components/Pagination.test.tsx

```diff
@@ -0,0 +1,148 @@
+    allowPagination: true
+  };
+  const mockedVerticalSearchState: VerticalSearchState = {
+    results: [mockedVerticalResults[0]],
+    resultsCount: 3,
+    verticalKey: 'vertical',
+    limit: 1,
+    offset: 0
+  };
+  mockVerticalSearchState(mockedVerticalSearchState);
+  render(<VerticalResults {...verticalResultsProps} />);
+  const paginationNavEl = screen.getByRole('navigation', { name: 'Pagination' });
+  expect(paginationNavEl).toBeDefined();
+  const totalPaginationButtons = within(paginationNavEl).queryAllByRole('button').length;
+  const numIconNavButtons = 2;
+  const numLabelNavButtons = mockedVerticalSearchState.resultsCount;
+  expect(totalPaginationButtons - numIconNavButtons).toEqual(numLabelNavButtons);
+});
+
+it('Pagination component is displayed with ellipses label', () => {
+  const verticalResultsProps: VerticalResultsProps = {
+    CardComponent: StandardCard,
+    allowPagination: true,
+    customCssClasses: {
+      leftIconContainer: 'leftNavButton',
+      rightIconContainer: 'rightNavButton'
+    }
+  };
+  const mockedVerticalSearchState: VerticalSearchState = {
+    results: [mockedVerticalResults[0]],
+    resultsCount: 5,
+    verticalKey: 'vertical',
+    limit: 1,
+    offset: 0
+  };
+  mockVerticalSearchState(mockedVerticalSearchState);
+  render(<VerticalResults {...verticalResultsProps} />);
+  const paginationNavEl = screen.getByRole('navigation', { name: 'Pagination' });
+  expect(paginationNavEl).toBeDefined();
+  const paginationButtons = within(paginationNavEl).queryAllByRole('button');
+  const numIconNavButtons = 2;
+  const numLabelNavButtons = 3;
+  // expected pagination layout with n results: [<] [1] [2] [...] [n] [>]
+  expect(paginationButtons.length).toEqual(numIconNavButtons + numLabelNavButtons);
+  expect(paginationButtons[0].classList.contains('leftNavButton')).toBeTruthy();
+  expect(paginationButtons[1].textContent).toEqual('1');
+  expect(paginationButtons[2].textContent).toEqual('2');
+  expect(paginationButtons[3].textContent).toEqual(`${mockedVerticalSearchState.resultsCount}`);
+  expect(paginationButtons[4].classList.contains('rightNavButton')).toBeTruthy();
+  expect(screen.getByText('...')).toBeDefined();
+});
+
+it('Pagination component navigation buttons trigger new search', () => {
+  const verticalResultsProps: VerticalResultsProps = {
+    CardComponent: StandardCard,
+    allowPagination: true
+  };
+  const mockedVerticalSearchState: VerticalSearchState = {
+    results: [mockedVerticalResults[0]],
+    resultsCount: 5,
+    verticalKey: 'vertical',
+    limit: 1,
+    offset: 0
+  };
+  mockVerticalSearchState(mockedVerticalSearchState);
+  const actions = spyOnActions();
+  render(<VerticalResults {...verticalResultsProps} />);
+
+  const paginationNavEl = screen.getByRole('navigation', { name: 'Pagination' });
+  expect(paginationNavEl).toBeDefined();
+
+  // navigate to the last results page
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
this is because of the "!" usage? Can we instead have something like `const mockedResultsCount = 5` and use that variable in `mockedVerticalSearchState` for resultsCount in line 123 and in the lines below?

Reply:
+1 this eslint disable should not be necessary

Reply:
Fixed!